unstable/wlr-exec-secure-unstable-v1.xml

```diff
@@ -0,0 +1,139 @@
+  <description summary="a protocol for executing clients with higher permissions">
+    This protocol allows unprivileged clients to ask the compositor to execute
+    a program which can access protected Wayland protocols.
+
+    The rationale for this protocol is that the execution environment of an
+    arbitrary client cannot be reliably verified by the compositor. Environment
+    variables like LD_PRELOAD may alter the runtime behavior of the client to
+    malicious ends. However, if the compositor creates the client socket and
+    exec's the client program itself, it can be sure of the client's execution
+    environment.
+
+    If a client is run with the WAYLAND_SOCKET environment variable set, it can
+    assume that the socket represents an authenticated Wayland connection and
+    it will be able to use any secure protocols it is afforded. If not, the
+    client will likely connect to the Unix socket specified by WAYLAND_DISPLAY,
+    but it will be an unauthenticated connection. If it needs authenticated
+    protocols to accomplish its basic tasks, it will likely wish to use
+    wlr_exec_secure_unstable_v1 to ask the compositor to execute itself again.
+    It's at the client's discretion to exit once the compositor runs the second
+    process or to wait for it to complete.
+  </description>
+
+  <interface name="zwlr_exec_manager_v1" version="1">
+    <description summary="interface for requesting execution handles">
+      Clients which want the compositor to execute another client should create
+      a zwlr_exec_handle_v1 and use it to configure and run the client.
+    </description>
+
+    <request name="exec_process">
+      <description summary="creates a wlr_exec_handle_v1 resource">
+        Creates an exec_handle for the requested command. The compositor can
+        choose to disallow this request for certain commands, in which case a
+        protocol error is issued. The command will be executed without a shell.
+      </description>
+      <arg name="handle" type="new_id" interface="zwlr_exec_handle_v1"/>
+      <arg name="command" type="array" summary="the command to execute"/>
+    </request>
+
+    <request name="destroy" type="destructor">
+      <description summary="destroy the manager">
+        All objects created by the manager will still remain valid, until their
+        appropriate destroy request has been called.
+      </description>
+    </request>
+  </interface>
+
+  <interface name="zwlr_exec_handle_v1" version="1">
+    <description summary="an executing or to-be-executed process">
+      This resource represents a process executed by the compositor, or a
+      process which will be executed by the compositor shortly.
+
+      The client should transfer any appropriate state, then send an exec
+      request and await the exit event.
+
+      Once the compositor receives an exec request, it should fork, close all
+      file descriptors (except for the transferred ones), then dup2 them to the
```

Comment:
>close all file descriptors

Including stdin, stdout, and stderr? Should they be set to /dev/null?
Obviously if they're explicitly set with fd, they should be set to that.

Reply:
We should probably specify that the behavior of stdin/stdout/stderr is unspecified if the client doesn't transfer them.